coral-trino/src/main/java/com/linkedin/coral/trino/rel2trino/transformers/SubstrIndexTransformer.java

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright 2023-2024 LinkedIn Corporation. All rights reserved.
+ * Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.coral.trino.rel2trino.transformers;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.apache.calcite.sql.SqlCall;
+import org.apache.calcite.sql.SqlIdentifier;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.SqlNumericLiteral;
+import org.apache.calcite.sql.fun.SqlStdOperatorTable;
+
+import com.linkedin.coral.common.calcite.CalciteUtil;
+import com.linkedin.coral.common.transformers.SqlCallTransformer;
+
+import static org.apache.calcite.rel.rel2sql.SqlImplementor.*;
+import static org.apache.calcite.sql.fun.SqlStdOperatorTable.*;
+
+
+/**
+ * This class transforms the substr indexing in the input SqlCall to be compatible with Trino engine.
+ * Trino uses 1-based indexing for substr, so the lowest possible index is 1. While other engines like Hive
+ * allow for 0 as a valid index.
+ *
+ * This transformer guarantees that starting index will always 1 or greater.
+ */
+public class SubstrIndexTransformer extends SqlCallTransformer {
+  @Override
+  protected boolean condition(SqlCall sqlCall) {
+    return sqlCall.getOperator().getName().toLowerCase() == "substr"
+        || sqlCall.getOperator().getName().toLowerCase() == "substring";
```

Comment:
Let's create a private final static immutable set for these 2 strings?

Reply:
Done!